gufe/tests/test_models.py

```diff
@@ -53,3 +53,45 @@ def test_invalid_constraint(value, good):
     else:
         with pytest.raises(ValueError):
             _ = OpenMMSystemGeneratorFFSettings(constraints=value)
+
+
+class TestFreezing:
```

Comment:
Current tests check that it works we do `settings.subsettings.item = ...`. Could you add a test that does `settings.subsettings = ...` ? (example: you attempt to replace the entire `thermo_settings` in one go).